benchmark/scripts/find_heap_bound.py

```diff
@@ -0,0 +1,188 @@
+        if self.suffix == target.suffix:
+            me = self.toBytes()
+            tar = target.toBytes()
+            res = tar * round(me / tar)
+            if res == 0:
+                res = tar
+            return HeapSize.from_bytes(res)
+        else:
+            return self
+
+    def __truediv__(self, div):
+        b = self.toBytes()
+        res = int(b / div)
+        return HeapSize.from_bytes(res)
+
+    def __mul__(self, m):
+        b = self.toBytes()
+        res = int(b * m)
+        return HeapSize.from_bytes(res)
+
+    def __add__(self, rhs):
+        return HeapSize.from_bytes(self.toBytes() + rhs.toBytes())
+
+    def __sub__(self, rhs):
+        return HeapSize.from_bytes(self.toBytes() - rhs.toBytes())
+
+    # Is this useful?
+    def approach(self, target, min_step):
+        """Binary search approach target with minimum step"""
+        me = self.toBytes()
+        t = target.toBytes()
+        s = min_step.toBytes()
+
+        diff = abs(me - t)
+        step = int(diff / 2)
+        if me > t:
+            res = me - step
+        else:
+            res = me + step
+        # Round to nearest multiple of s for prettiness
+        res = s * round(res / s)
+        # And saturate at s
+        res = s if res == 0 else res
+        return HeapSize.from_bytes(res)
+
+
+    @classmethod
+    def from_str(cls, s: str):
+        regex = '(\d+)([kKmMgG])?'
+        m = re.match(regex, s)
+        if m:
+            suffix = m.group(2)
+            if suffix is None:
+                return HeapSize(int(m.group(1)), "")
+            else:
+                return HeapSize(int(m.group(1)), suffix.upper())
+        else:
+            raise Exception("Invalid Heap Size {}!".format(s))
+
+    @classmethod
+    def from_bytes(cls, b: int):
+        if b % cls.G_FACTOR == 0:
+            return HeapSize(round(b / cls.G_FACTOR), "G")
+        if b % cls.M_FACTOR == 0:
+            return HeapSize(round(b / cls.M_FACTOR), "M")
+        if b % cls.K_FACTOR == 0:
+            return HeapSize(round(b / cls.K_FACTOR), "K")
+        return HeapSize(round(b), "")
+
+
+    def __str__(self):
+        return str(self.value) + self.suffix
+
+
+def parseargs():
+    parser = argparse.ArgumentParser("Bound JVM Heap Size")
+    parser.add_argument("--verbose", "-v", action="count", default=0,
+                        help="Increase verbosity level (cumulative)")
+    parser.add_argument("args", type=str, nargs="+",
+                        help="Arguments to JVM, include classpath and main")
+    parser.add_argument("--start-size", type=str, default="4G",
+                        help="Starting heap size (default: 4G)")
+    parser.add_argument("--min-step", type=str, default="100M",
+                        help="Minimum heap size step (default: 100M)")
+    parser.add_argument("--java", type=str, default="java",
+                        help="Java executable to use (default: java)")
+    return parser.parse_args()
```

Comment:
You can avoid having the `default` strings if you use:

```python
parser= argparse.ArgumentParser(
  prog="Bound JVM Heap Size",
  formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```

Reply:
I looked for something like this but didn't find it. This is great.